File: backend/src/com/ffhs/model/Game.java
package com.ffhs.model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

/**
 * Entity class for Game
 */
@Entity
public class Game {
    @Id
    @GeneratedValue(strategy= GenerationType.AUTO)
    private int gameId;

    private int gameIdInTournament;

    private int tournamentId;

    private String p1Id;

    private String p2Id;

    private int p1Score;

    private int p2Score;

    /**
     * @return int returns the gameId
     */
    public int getGameId() {
        return gameId;
    }

    /**
     * sets the gameId
     * @param gameId int
     */
    public void setGameId(int gameId) {
        this.gameId = gameId;
    }

    /**
     * @return int returns the gameId withing the tournament
     */
    public int getGameIdInTournament() {
        return gameIdInTournament;
    }

    /**
     * sets the gameId within the tournament
     * @param gameIdInTournament int
     */
    public void setGameIdInTournament(int gameIdInTournament) {
        this.gameIdInTournament = gameIdInTournament;
    }

    /**
     * @return int returns the tournamentId
     */
    public int getTournamentId() {
        return tournamentId;
    }

    /**
     * sets the tournamentId
     * @param tournamentId int
     */
    public void setTournamentId(int tournamentId) {
        this.tournamentId = tournamentId;
    }

    /**
     * @return String returns the guid of player1
     */
    public String getP1Id() {
        return p1Id;
    }

    /**
     * sets the guid of player1
     * @param p1Id String
     */
    public void setP1Id(String p1Id) {
        this.p1Id = p1Id;
    }

    /**
     * @return String returns the guid of player2
     */
    public String getP2Id() {
        return p2Id;
    }

    /**
     * sets the guid of player2
     * @param p2Id String
     */
    public void setP2Id(String p2Id) {
        this.p2Id = p2Id;
    }

    /**
     * @return int returns the score of player1
     */
    public int getP1Score() {
        return p1Score;
    }

    /**
     * sets the score of player1
     * @param p1Score int
     */
    public void setP1Score(int p1Score) {
        this.p1Score = p1Score;
    }

    /**
     * @return int returns the score of player1
     */
    public int getP2Score() {
        return p2Score;
    }

    /**
     * sets the score of player1
     * @param p2Score int
     */
    public void setP2Score(int p2Score) {
        this.p2Score = p2Score;
    }
}

File: frontend/src/modules/tournaments/stores/models/game.ts
import {IGame} from "../../../common/apiTypings";
import {action, observable} from "mobx";

export default class Game {
    public static createFromTournamentDto(dto: IGame): Game{
        const m = new Game();
        m.gameId = dto.gameId;
        m.gameIdInTournament = dto.gameIdInTournament;
        m.tournamentId = dto.tournamentId;
        m.p1Id = dto.p1Id;
        m.p2Id = dto.p2Id;
        m.p1Score = dto.p1Score;
        m.p2Score = dto.p2Score;

        return m;
    }

    @observable public gameId: number | undefined;
    @observable public gameIdInTournament: number | undefined;
    @observable public tournamentId: number | undefined;
    @observable public p1Id: string | undefined;
    @observable public p2Id: string | undefined;
    @observable public p1Score: number | undefined;
    @observable public p2Score: number | undefined;


    @action
    public update(dto: IGame){
        this.gameId = dto.gameId;
        this.gameIdInTournament = dto.gameIdInTournament;
        this.tournamentId = dto.tournamentId;
        this.p1Id = dto.p1Id;
        this.p2Id = dto.p2Id;
        this.p1Score = dto.p1Score;
        this.p2Score = dto.p2Score;
    }
}